apis/keda/v1alpha1/triggerauthentication_types.go

```diff
@@ -282,6 +285,30 @@ type AzureKeyVaultCloudInfo struct {
 	ActiveDirectoryEndpoint string `json:"activeDirectoryEndpoint"`
 }
 
+type GCPSecretManager struct {
+	Secrets []Secrets `json:"secrets"`
+	// +optional
+	Credentials *GCPCredentials `json:"gcpCredentials"`
+	// +optional
+	PodIdentity  *AuthPodIdentity `json:"podIdentity"`
+	GCPProjectID string           `json:"gcpProjectID"`
```

Comment:
Maybe it's a stupid question, but, can't we infer this value from credentials or pod identity? AFAIR, credentials json contains the project, and we can query the project to metadata api or reading the (not GCP) workload webhook

Reply:
You're right, for a credentials JSON file it is certainly possible to parse the project ID from it, however, for pod identity I am not sure if we can get the project ID (or how to get it). That was the intuition behind making the field mandatory (for both credentials JSON input and pod identity based authentication), to avoid confusion.

If you can share an approach for getting the project ID when using pod identity, I will remove this field.

Reply:
Using PodIdentity you can get it too using metadata client or another already set env (metadata is available if your ar inside GCP and the env is available if you are outside GCP because the mutating webhook adds it).
You have an example here: https://github.com/kedacore/keda/blob/main/pkg/scalers/gcp_stackdriver_client.go#L84
If you have any doubt just ask me :)

Reply:
Hey @JorTurFer, I have made the requisite changes PTAL, TY!